windows/src/buildtools/sentrytool/Keyman.System.AddPdbToPe.pas

```diff
@@ -0,0 +1,253 @@
+unit Keyman.System.AddPdbToPe;
+
+interface
+
+uses
+  System.AnsiStrings,
+  System.Classes,
+  System.SysUtils,
+  Winapi.ImageHlp,
+  Winapi.Windows;
+
+function AddDebugID(s: TMemoryStream; guid: TGUID; var codeId: string): Boolean;
+
+var
+  AddPdbToPe_Debug: Boolean = False;
+
+implementation
+
+function RVA2Offset(nt: PImageNtHeaders; rva: DWORD): DWORD;
+var
+  section: PImageSectionHeader;
+  sectionIndex: Integer;
+begin
+  section := PImageSectionHeader(PByte(nt) + sizeof(IMAGE_NT_HEADERS));
+
+  for sectionIndex := 0 to nt.FileHeader.NumberOfSections - 1 do
+  begin
+    {
+        Check if the RVA is within the virtual addressing space of the section
+        Make sure the RVA is less than  the VirtualAddress plus its raw data size
+        IMAGE_HEADER_SECTION.VirtualAddress = The address of the first byte of
+            the section when loaded into memory, relative to the image base. For
+            object files, this is the address of the first byte before
+            relocation is applied.
+        Our ImageBase is 0, since we aren't loaded into actual memory
+    }
+    if (rva >= section.VirtualAddress) and (rva < section.VirtualAddress + section.SizeOfRawData) then
+    begin
+      Exit(section.PointerToRawData + rva - section.VirtualAddress);
+    end;
+    Inc(section);
+  end;
+
+  Result := 0;
```

Comment:
So, we don't throw any sort of error if our assumptions are invalid here?  Given what we're dealing with, I think it's better that we throw errors / crash the tool for the `else` case - where `rva` isn't within the expected bounds.

At minimum, some sort of warning that helps us to diagnose issues should they arise in the future.  I mean, we have `debug_writeln` and `error_writeln` defined below; we should use the latter here.